base/linalg/dense.jl

```diff
@@ -29,6 +33,41 @@ function scale!(X::Array{T}, s::Real) where T<:BlasComplex
     X
 end
 
+
+function isone(x::StridedMatrix)
+    m, n = size(x)
+    m != n && return false # only square matrices can satisfy x == one(x)
+    if m < div(ISONE_CUTOFF * m * m, sizeof(x))
```

Comment:
I'm confused. Morally, this checks is more or less equivalent to `m*sizeof(x) < CUTOFF*m^2 <=>sizeof(x)/m < CUTOFF`. Assuming `eltype(x)` is a concrete type, `sizeof(x)/m` is the size of an element, but the test should depend on the size of the whole matrix (which is what you did initially when you had `if m < TRIA_CUTOFF` (with an implicit fixed `sizeof(eltype(x))`).